Web/Harvest/API.hs

```diff
@@ -65,35 +66,32 @@ getUsers_ :<|> getTimeEntries_ = client (Proxy :: Proxy HarvestAPI)
 
 -- | Get list of all users for specific account.
 
-getUsers :: MonadIO m
-  => Manager           -- ^ HTTPS manager
-  -> Credentials       -- ^ Credentials
-  -> m (Either ServantError [User]) -- ^ Result of request
-getUsers = runHarvestQuery getUsers_
+getUsers :: Manager                         -- ^ Http Manager
+         -> Credentials                     -- ^ Credentials
+         -> IO (Either ServantError [User]) -- ^ Result of request
+getUsers manager creds =
+  let env = ClientEnv manager (getBaseUrl creds)
+  in runClientM (runHarvestQuery getUsers_ creds) env
 
 -- | Get time entries for specific date and user.
 
-getTimeEntries :: MonadIO m
-  => Manager           -- ^ HTTPS manager
-  -> Credentials       -- ^ Credentials
-  -> Day               -- ^ Date of interest
-  -> UserId            -- ^ User id
-  -> m (Either ServantError TimeEntries)
+getTimeEntries :: Manager           -- ^ Http Manager
+               -> Credentials       -- ^ Credentials
+               -> Day               -- ^ Date of interest
+               -> UserId            -- ^ User id
+               -> IO (Either ServantError TimeEntries)
 getTimeEntries manager creds date uid =
-  runHarvestQuery (getTimeEntries_ day year (Just uid)) manager creds
-  where (day, year) = getDayAndYear date
+  let env = ClientEnv manager (getBaseUrl creds)
+      (day, year) = getDayAndYear date
+  in runClientM (runHarvestQuery (getTimeEntries_ day year $ Just uid) creds) env
 
 -- | A helper to run a query against Harvest API.
 
-runHarvestQuery :: MonadIO m
-  => (BasicAuthData -> Manager -> BaseUrl -> ClientM a) -- ^ Query function
-  -> Manager           -- ^ HTTPS Manager
-  -> Credentials       -- ^ Credentials
-  -> m (Either ServantError a) -- ^ The result
-runHarvestQuery action manager Credentials {..} = liftIO $ do
-  let host     = BC8.unpack credentialsAccount ++ ".harvestapp.com"
-      authData = BasicAuthData credentialsUsername credentialsPassword
-  runExceptT (action authData manager (BaseUrl Https host 443 ""))
+runHarvestQuery :: (BasicAuthData -> ClientM a) -- ^ Query function
```

Comment:
Could we replace it by its type synonym?
`type Query a = BasicAuthData -> ClientM a`